segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -253,7 +278,11 @@ protected void doStart() {
     }
 
     private CompletableFuture<Void> initializeSecondaryServices() {
-        this.storage.initialize(this.metadata.getContainerEpoch());
+        try {
+            initializeStorage();
+        } catch (Exception ex) {
```

Comment:
we are not dealing with InterruptedException here

Reply:
The caller handles it.

Reply:
You are returning a CompletableFuture so the caller probably won't run in this thread.
As usual we should call Theead.currentThread().interrupt()
otherwise we are going to mask that flag

Reply:
Not sure why we need to do this.
This task was executed on this thread and received thread interrupted exception - which we are returning as failed future.

This thread is now supposed to pickup some other work and complete it. What happens if we don't call Thread.currentThread().interrupt() ?

Reply:
you can't know who the caller is, because we can refactor this code in the future
and as when you catch an InterruptedException you reset the 'interrupted' flag on the Java thread,
if you do not set again that flag the caller won't be notified about this InterruptedException

even if I am not a fan of InterruptedException and how it works IMHO we should handle it consistently (using Exceptions#handleInterrupted)

Reply:
We do not do this anywhere else in the code. If this must be changed, then we should do it throughout the codebase, not just here.

@sachin-j-joshi I would suggest to clean this up and add a follow-up issue that will add a helper in `Futures` to do this.

Reply:
Created a tracking issue https://github.com/pravega/pravega/issues/4986